### PR TITLE
Fix macOS LAMMPS plugin loading issue

### PR DESCRIPTION
**Pull request by njzjz**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Patch https://github.com/deepmodeling/deepmd-kit/pull/3473.

### CODE REVIEW

**Comment by njzjz** [comment]
@conda-forge-admin, please rerender

**Comment by njzjz** [comment]
@conda-forge-admin, please rerender

**Comment by njzjz** [comment]
@conda-forge-admin, please rerender

**Comment by chazeon** [comment]
Hi @njzjz, see this issue https://github.com/conda-forge/conda-forge.github.io/issues/2102, it seems that  you will also need to add `c_stdlib` and `c_stdlib_version` to `conda_build_config.yaml` to make it work.

**Comment by njzjz** [comment]
> Hi @njzjz, see this issue https://github.com/conda-forge/conda-forge.github.io/issues/2102, it seems that  you will also need to add `c_stdlib` and `c_stdlib_version` to `conda_build_config.yaml` to make `{{ stdlib("c") }}` work.

The global pining has contained it: https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/3c7d60f56a8cb7d1b8f5a8da0b02ae1f1f0982d7/recipe/conda_build_config.yaml#L11-L24
The issue seems to be due to an old version  of conda-build: https://github.com/conda-forge/conda-build-feedstock/pull/220

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,6 +33,7 @@ build:
 
 requirements:
   build:
+    - {{ stdlib("c") }}
```
Comment (jakirkham) [inline recipe/meta.yaml:36]: We are still working through adding `{{ stdlib("c") }}`

For now would just add `sysroot` and pin it to `2.17`. Would re-render after to propagate changes through the feedstock

```suggestion
    - sysroot_{{ target_platform }} 2.17  # [linux]
```

**Review by jakirkham — COMMENTED** [review]
Hey Jinzhe, sorry for the trouble here

Added a suggestion below

Also may want to add this to `conda-forge.yml`

```yaml
os_version:
  linux_64: cos7
  linux_aarch64: cos7
  linux_ppc64le: cos7
```

Then would re-render for these to take effect

**Comment by njzjz** [comment]
@conda-forge-admin, please rerender

**Comment by njzjz** [comment]
@conda-forge-admin, please rerender

**Pull request merged**